### ISSUE TITLE
T1200XE differences

### ISSUE THREAD

**Issue by thetechknight** [fust/T1600-PSU#3]
I love what you did with this project and I am working on a T1200xe which is slightly different so I made a comment on your YouTube video but just in case you didn't get it or if it got moderated, ill post here: 

The T1200xe has a dead power supply from cap juice as they all do anymore, so I was referring to your schematic since its the only one out there in the lineup of Toshibas. It is very similar but it is also different. Unfortunately, its different in a way that matters so I was hoping you could help clarify. 

This one, there is IC14, 15, and 17 which are completely eaten away. They all appear to have the marking "C5" which your T1600 doesn't show in the schematic so they mustn't have been used in that model. SMD Codebook points to an LDO regulator, but there's no suffix markings to indicate voltage so I am having a hard time believing this is the correct information and its likely something else. They look like a SOT-23 with 5 legs instead of 3. 

Also IC13 which appears to be your IC15, part of the RAM power control is also eaten away and its not really shown in your schematic what part it is for replacement, perhaps its proprietary. 

Any thoughts? Thanks!

**Comment by fust** [comment]
I'll reply here as it's a bit easier than the YouTube comments.

The T1200XE PSU is quite a bit different to the T1600. Although most all T1000-series use roughly the same architecture, the way the DC-DC converters are controlled varies quite a bit.
I wouldn't feel comfortable comparing them one-on-one or even using this schematic as a reference for anything other than the DC-DC converters.

It's been a while since I lasted worked on the PSU in my T1200XE and hard to tell without the board or some good picture but I think I know the ICs you're talking about.
They're the ones on the underside of the board, right below the large capacitors towards the left side, correct?
IIRC those packages are indeed 5-pin SOT23, which are actually called SOT-23-5 but I'll have to dig out my T1200XE to verify check the markings.
A quick code search for "C5" in that package came up with FMC5A or UMC5N, which are NPN/PNP transistor pairs.
That makes more sense to me than LDOs as they're very near the DC-DC converter MOSFETs and are probably a single-package solution to drive the gates.

IC15 in the T1600 schematic is a 5V linear voltage regulator that powers the (S)RAM when the machine is "suspended", it can probably be replaced by for example the LM7805 or similar.
Be _very_ sure that IC13 in the T1200XE is the same thing before doing so though.

**Comment by thetechknight** [comment]
Thanks for taking the time and responding, appreciate the update! 

Yeah, the voltage regulator angle just wasn't making sense to me on those 5 pin SOTs, a transistor does make sense but on the PCB they are referring to them as ICs. I guess because it has more than one transistor, they decided to call it an IC. Looking at the datasheet for the transistor you specified, There's a pin on the PCB thats just unconnected on all 3, which is critical to that 2nd transistor so that doesn't add up either. 

The marking I think is a C5. the cap juice has eaten away the silkscreen layer and i tried to scrub the components to reveal the C5. so I THINK it is a C5. And they are clustered over next to those big caps on the right side if you are looking at the back of the PCB. They all appear to be part of that SRAM power circuit I think. 

I had to do a trace diagram because there are many many many open traces. its pretty bad. Also one of the SOT89 mosfets are completely blown apart as well. 

![image](https://github.com/fust/T1600-PSU/assets/22461464/b72089c5-17c8-440f-b930-8c736c54c852)

**Comment by thetechknight** [comment]
Heres some pictures...

Here is a shot from before I cleaned the PCB:
![image](https://github.com/fust/T1600-PSU/assets/22461464/3096dcf9-4447-4369-9197-c0571e6a6219)

Here is the best I could do cleaning off the semiconductors to get the marking codes. They are just completely eaten off:
![image](https://github.com/fust/T1600-PSU/assets/22461464/5d33171f-bc62-46e5-8276-1c288c7be74a)

Better marking of IC13:
![image](https://github.com/fust/T1600-PSU/assets/22461464/cb487d5e-c81c-48f6-94c5-8052008fa595)

**Comment by fust** [comment]
Yikes, that looks really bad.

My money would be more on damaged traces than semiconductors.
In my experience only the large MOSFETs and diodes are prone to failure (due to bad caps), the rest is usually fine but YMMV.

I found my T1200XE and took some pictures of IC13 through 17, you were spot on on the C5 marking:
![ic13](https://github.com/fust/T1600-PSU/assets/1225601/c2d3d8d8-dd92-4bbc-a887-fa08ce7355d5)
![c51](https://github.com/fust/T1600-PSU/assets/1225601/58431dbe-30f3-4776-b4bd-dd97041e63d5)
![c52](https://github.com/fust/T1600-PSU/assets/1225601/f97eebbc-0858-474f-80b4-4d44ebb47f2b)
Unfortunately I can't find anything else matching this marking and package through a quick search.

My T1200XE is working fine apart from a hit-and-miss harddrive so I could cross-check some things if need be.

**Comment by thetechknight** [comment]
Ahh glad thats confirmed. 

Yeah, I need to find out what C5 actually is before I can move forward because mine are eaten away. the pins are gone once I got it all cleaned up. Or, what they do in the circuit to see if they can be patched out/omitted/modified.

**Comment by thetechknight** [comment]
Hey, is it possible you can figure out from yours where these pads are connected? They dont appear to go anywhere on mine but the traces could simply be eaten away to not being visible. I found a few like that. The transistor pair part number you mentioned requires this connection to be accurate. 

![image](https://github.com/fust/T1600-PSU/assets/22461464/45159983-b742-4e16-9de2-a84120eca66a)

**Comment by fust** [comment]
I couldn't measure anything so I've removed IC14 to see if anything was going on underneath but it seems this pin is just not connected by design. The other 2 will probably be identical so I didn't bother removing those to check.

I'm feeling a bit conflicted about this but that would mean it's not an FMC5A or UMC5N.
It is related to the switching circuitry as it is connected to the MOSFET gate, 5V regulator, GND and the main controller IC.
Though it looks like a good candidate I just fail to see how it should work at all with that component.

**Comment by thetechknight** [comment]
Yeah then it must be something else. What though is the question...

Wonder if its the mosfet driver ICs then?

**Comment by fust** [comment]
That's exactly what I meant by "part of the switching circuitry" and also why the transistor pair IC's are a good candidate.
On the T1600 (and the T1000, T1000LE/XE and probably some others) this is handled by discrete transistors but it looks like it was integrated into a single package for at least a couple of power rails on the T1200XE.

I'd probe the IC's with a scope to see what exactly is going on but it's impossible to reach the components when the PCB is mounted in the computer and I haven't figured out how to power up the PSU without the rest of the machine connected.
Should be possible but unfortunately it's not as straight forward as with the T1600 which "just works" stand-alone.

If I have some time this week I'll see if I can think of something, if and when I find anything I'll get back to you.

**Comment by thetechknight** [comment]
I got another PCB from a parts T1200xe that wasnt quite in as bad of condition. So between the two PSU boards, I was able to merge one good PSU board. 

So i patched up all the broken traces i could find, and then a fresh set of caps. 

Only thing I could get is a blinking red LED. Nothing else. None of the converters are even attempting to come up. So i looked at your schematics again, and it appears the MCU and gate array section appears to be close if not identical so I started doing some checks. I get a clock and reset pulse along with 5V to that control MCU but the MCU and gate array dont appear to be doing anything. Cant even get any PWM Drive signals out at all. 

So something is giving up before the MCU attempts to bring up the voltages. Only trouble is, I dont know what the MCU is looking for to satisfy a condition before it brings up the voltages. Or maybe the MCU itself is dead. Thoughts? 

Only breaks i could find: There were a number of them though
![image](https://github.com/fust/T1600-PSU/assets/22461464/431825b2-aa93-4226-b0a7-874ebb4bd9ab)

My patch work: 
![image](https://github.com/fust/T1600-PSU/assets/22461464/43c1e5a6-98b4-40b7-8f7a-7a8abcae9d19)

It did have its fair share of leakage too:
![image](https://github.com/fust/T1600-PSU/assets/22461464/b0edcd1e-083d-4ca4-92ff-170b6b3c6f45)

**Comment by fust** [comment]
Just to be sure: I assume you're testing and probing the board when it's installed in the machine? As I mentioned earlier it flat out won't work when it's not connected to the rest of the computer.
I haven't had the chance to tinker with it and figure out a way to make it work standalone, I've been quite busy unfortunately.
One thing to note is that if the PSU attempts to power up the PWM signals will only be present for a very short amount of time, a digital oscilloscope in single-shot mode is absolutely invaluable here.

If it's that similar to the T1600 one thing springs to mind (TL;DR below):
On the T1600 the two main microprocessors work in tandem: IC12 is the charge controller and also handles the sensing of the (standby) RAM voltage and main input voltage. It receives it's clock from X2 which is 3-pin package containing a crystal and 2 ceramic capacitors.
When the power button is pressed IC12 powers on IC13 (and keeps it powered for a few seconds after the PSU shuts down for whatever reason). It receives it's clock from X1 through the gate array (probably for buffering).

When X2 (IC12 clock) isn't working the PSU does absolutely nothing. Seeing you're getting the flashing red LED (which, on the T1600, is controlled by IC12) that part seems to be executing code.
When X1 or the gate array isn't working OR the transistor that powers the regulator or the regulator itself for IC13 isn't working there will be no PWM, which seems to be what's happening on your board.

TL;DR: Check to see if the microprocessor that controls the PWM signals is receiving power and a clock.
If that all looks good, there are a couple of lines between the two processors (probably serial communication lines, haven't looked in to it) that need to be good as well.

**Comment by thetechknight** [comment]
Yeah I am testing it while its in the machine. when pressing the power button, 5V gets applied to the 2nd MCU for a brief bit of time, yes. 

PWM signals never come up. 5V is applied to the gate array and the secondary MCU. the RESET signal on that secondary MCU gets a quick burst too... 

the clock signal is present to both MCUs. The clock going to the MCU in question is 10Mhz, which appears to be coming from the gate array. Gate array is seeing a 20Mhz clock. 

The PWM signals coming out of the gate array just stay pegged at 5V when power is attempted. theres no burst or anything of anything. 

Last ditch effort, I swapped the MCU and Gate array from the other power supply board and no dice. 

None of the rails are getting any hints of a burst of voltage during the power on attempt. seems to last about 1 second before it flashes. While its flashing, if i leave the PSU sit for 10 seconds or so, the 5V will drop itself back off until I hit the power button again and the cycle repeats

**Comment by fust** [comment]
That's a pretty tough one to diagnose remotely.
Waiting for the 5V to drop off on the second MCU is critical before re-attempting, it needs to be powercycled completely (but you probably figured that out already).

Another thought: does the T1200XE also have an internal 18V bus like the T1600 does?
I recall that that failed on my first T1600, the SMD MOSFET failed completely and had a large hole in it so it was quite obvious to spot.
Without the internal 18V bus (or it's sensing circuitry) the MCUs wouldn't do anything.
It doesn't have to be a failed MOSFET ofcourse but I'd say it's far more likely than a failed diode or inductor.

The MCUs are pretty well protected with all the buffering, I'd be surprised if they failed.
A final thing that springs to mind is a faulty uPA1600, that would certainly throw a wrench in things.
Fortunately those are fairly easy to test out-of-system as they're just inverter arrays.

**Comment by thetechknight** [comment]
Thats where the circuitry differs. This one doesnt appear to have an 18V supply but I do not know. 

It appears to have at least 2 converters since I see two inductors, but it also has a transformer so maybe 2 converters and 1 flyback transformer style switcher. 

If it was the uPA1600, I would still have the drive signals from the gate array going into it, at least briefly, no?

**Comment by fust** [comment]
In that case I'd check that the MOSFETs are getting power to either the drain or the source (depends on the type of MOSFET), it should be getting _at least_ 12V when plugged in to the AC brick but I would think it's more.
The T1600 uses the internal 18V bus as a stable baseline, so it doesn't matter if it's powered from the wall or from battery. I wouldn't be surprised if the T1200XE uses a similar scheme.

It could still be that a dead uPA1600 causes the symptoms you're seeing, if it failed shorted to ground on all gates you wouldn't see any signals at all. Not very likely IMHO but certainly not impossible.

I'll see if I can find some time to probe around the board a bit tomorrow, it's getting a bit too late for that over here.

**Comment by thetechknight** [comment]
No rush! That would be awesome if you could do that, since I do not have a working board to make a comparison chart. 

Id love to have some sort of voltage chart or waveform chart for the MCUs and gate array, just so it could help me narrow down which signal might be the faulty signal that is holding me up. 

I have my HP Scope and I probe up with that during initial power-on to see if I get any hint of a waveform and I do not.

**Comment by fust** [comment]
I've made some measurements on the board.
Apparently it's able to run standalone as long as the small board containing 2 switches and an NTC is connected to PJ3.
As far as I can tell only the NTC is strictly necessary, without it I only get a blinking red LED.
The center wire of the connector goes straight to ground, the red wire (pin 1) goes through R3 (10K) to pin 2 of IC1. Pin 3 shouldn't matter.

Anyway, I've taken some voltage measurements that might be helpful. 'Scope shots will have to come later as I seem to have misplaced the USB flashdrive I use for that.

Note: These are all unloaded voltages (running standalone)

With the PSU turned off:
Q1 drain is at ~3V, Gate & Source is at 0V
Q6 Gate & Source both have 5V, Drain 0V
Q7 has 12V across all pins
Q8 has ~ -3V on it's gate, 12V on Source & Drain
L4 is at 12V
L1 is at 0V
On the underside:
Q14 has 12V on it's center pin (pin 2), 0 on 1&3
Q21 has 12V on pin 1, 0V on 2&3
Q22 is at 0V on all pins
Q27 has 0V on pin 1, 5V on pins 2&3
Q28
Q29 has 12V on pin 2, 0V on 1&3

With the PSU running:
Q1 Gate has PWM, Drain is at 12V, Source is at 5V
Q6 Source & Drain 5V, Gate 0V
Q7 12V across all pins
Q8 Gate 0V, Drain & Source 12V
L4 is at 12V
L1 is at 5V
On the underside:
Q14 has ~ 1.5V on pin 1, 12V on pin 2, 0V on pin 3
Q21 has 0V on pin 1, 22V on pin 2, 0V on pin 3
Q22 has PWM on pin 1, 12V on pin 2, 0V on pin 3
Q27 has 5V across all pins
Q28 has around -9V on all pins
Q29 has 12V on pin 2, 0V on 1&3
"22" has PWM on pin 1, 5V on 2&3
"27" has -9V on pin 1, 0V on pin 2, PWM on pin 3

Q16 through Q19 are set up as 2 pairs of gate drivers for PWM.
"35", "36" and "37" have PWM signals present so they have to do something with it as well.
IC5 seems to handle most of the PWM signal buffering, might be coming from IC3 as it has a lot of squarewaves on it's pins but haven't cheked.
Q2, Q3 and Q4 seem to have something to do with the 12V bus.
Q2 doesn't have PWM signals present so it is just used as a switch, Q2 and Q3 do have PWM signals.

Hope that this is enough to help you for now, If I have the time I can make some further measurements and make a couple of 'scope shots.

**Comment by thetechknight** [comment]
Awesome, thanks for this! 

Can you provide the microcontroller/gate array state on its pins on/off? Because I think thats whats going on is the microcontroller isnt happy about something.

**Comment by fust** [comment]
You'd have to be a bit more specific on which pins are of interest, it's over 80 pins between the 3 of them with a lot of different types of signals.
Can you verify the connection to PJ3 is good and the NTC is reading sensible values? It's at ~11K for me with an ambient temperature of 19C (66F).

**Comment by thetechknight** [comment]
That's just it, I don't know enough about the architecture of the PSU to be specific :-( 

If I were to narrow it down, I suppose the pins responsible for feedback or allowing the PSU to come on/enable the PWMs? 

and yeah i had to have that little side board plugged in or the battery orange light would flip on for 1 second and go out with a blinking red light. But I can verify the NTC, although I dont recall seeing one?

**Comment by fust** [comment]
That'll take me a while as it requires some reverse engineering and my free time is unfortunately very limited the coming week.
You can take a head start on it yourself though, the trick I like to use is to trace backwards from the board-to-board interconnect.
The pinout is available in the [T1200XE maintenance manual](https://www.minuszerodegrees.net/manuals/Toshiba/Other/Toshiba%20T1200XE%20-%20Maintenance%20Manual.pdf) on page APP-14 (page 132 in the PDF) , the output voltages are on page 1-15.
Pin 1 on the connector is marked with a white dot, the pins on the PCB side are a bit hard to wrap your head around, see page 6 of [the datasheet for the connector](https://multimedia.3m.com/mws/media/218514O/3mtm-050-in-bm-plg-050thru-hole-strt-ra-w-ret-clps-ts2046.pdf) on how it is laid out.

Working backwards from there you should end up at a capacitor which is the output tank for the DC-DC converters.
From there the DC-DC converter components are easily identified which will get you to the lines driving them, usually through a pair of NPN/PNP driver transistors.
The sense lines are usually created with a voltage divider between the output lines and ground and the center tap sent off to the MCU.

On the small side board there should be 2 switches and an NTC: The lid closed switch is mounted vertically, a "battery present" switch which is a small tactile at 90 degrees and the NTC. The last one is a small blue blob just above the connector and has one lead covered with a silicon tube.

**Comment by thetechknight** [comment]
Does the battery have to be present for this PSU to run? or can it run directly from the 12V input. Also does the RAM battery have to be present and good? 

I just want to eliminate variables, thanks.

**Comment by fust** [comment]
The batteries do not have to be present. The PSU will run fine on just the 12V AC brick.
The only requirement is the small board containing the 2 switches and NTC to be connected and the NTC to be working.
That's how I ran it to get the voltages earlier.

**Comment by thetechknight** [comment]
oh awesome! so that eliminates a couple variables. The fact that itll fire up without the motherboard makes things far easier.
